### DISCUSSION TITLE
Kong route versioning

### DISCUSSION THREAD

**Discussion by namanjain98** [Kong/kong#13593] [Help]
I have deployed a version of a proxy in my Kong cluster, which is currently working well. Now, I want to introduce some changes to the proxy and enable a canary traffic deployment. Specifically, I want version 1 of the proxy, which was previously deployed, to continue receiving 90% of the traffic, while version 2, which includes the new changes, should receive the remaining 10% of the traffic.

Example:

I initially deployed a proxy named proxy_1 (version 1).
After making changes to the Lua code, I deployed the updated proxy as proxy_1 (version 2).
Now, I need to configure traffic routing such that x% of the traffic is directed to proxy_1 (version 1), while (100-x)% of the traffic goes to proxy_1 (version 2).

**Comment by chronolaw** [comment]
We have a canary plugin (https://docs.konghq.com/hub/kong-inc/canary/) for this, but it only works for Kong Enterprise.
